identity.go

```diff
@@ -409,36 +442,59 @@ func (id *ID) queueReload(closeNotify <-chan struct{}) {
 func LoadIdentity(cfg Config) (Identity, error) {
 	id := &ID{
 		Config: cfg,
-		cert:   &tls.Certificate{},
 	}
 
 	var err error
-	id.cert.PrivateKey, err = LoadKey(cfg.Key)
-	if err != nil {
-		return nil, err
+
+	var defaultKey crypto.PrivateKey
+	if cfg.Key != "" {
+		var err error
+		defaultKey, err = LoadKey(cfg.Key)
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	if idCert, err := loadCert(cfg.Cert); err != nil {
-		return id, err
-	} else {
-		id.cert.Certificate = make([][]byte, len(idCert))
-		for i, c := range idCert {
-			id.cert.Certificate[i] = c.Raw
+	// CA bundle is optional, but can be used to fill in the client cert chain
+	if cfg.CA != "" {
+		if id.ca, id.caPool, err = loadCABundle(cfg.CA); err != nil {
+			return nil, err
+		}
+	}
+
+	if cfg.Cert != "" {
+		if defaultKey == nil {
+			return nil, errors.New("no key specified for identity cert")
+		}
+
+		id.cert = &tls.Certificate{
+			PrivateKey: defaultKey,
+		}
+
+		if idCert, err := loadCert(cfg.Cert); err != nil {
+			return nil, err
+		} else {
```

Comment:
Isn't this `else` clause unnecessary, as the `if` above is a guard clause?

Reply:
Kind of. idCert only exists in the context of the if/else block. If I remove the else, then it won't compile. I could remove the inline of the if, but then that adds an extra line as well.